ops.py

```diff
@@ -11,63 +14,123 @@ def code(self):
     def __str__(self):
         pass
 
+@dataclass(frozen = True)
+class Data(Expr):
+    name : str
+
+    def get_key(self):
+        return self.name
+
+    def code_name(self):
+        return f"data__{self.name}"
+
+    def code(self):
+        return self.code_name()
+
+@dataclass(frozen = True)
+class Index(Expr):
+    names : Tuple[str]
+    
+    def get_key(self):
+        return self.names
+    
+    def code_name(self):
+        # This code runs:
+        return f"index__{'_'.join([name.__str__() for name in self.names])}"
+        # But I think it should be:
+        # return f"index__{'_'.join(self.names)}"
```

Comment:
I left a comment [here](https://github.com/bbbales2/regressions/pull/3#discussion_r727733875) regarding this. Originally I wanted to allow expressions as indices but with the current compiler state it would make it tricky to keep track of the index names. So I just went with the simple way - only allow literals as indices, which can just be strings.